From a7f86e56017099c8e428af73a778eb08d8df092e Mon Sep 17 00:00:00 2001
From: Weifeng Liu <weifeng.liu@intel.com>
Date: Fri, 12 Jan 2024 18:28:53 +0000
Subject: [PATCH 08/14] ivshm: Fix double release bug

We shall not call release function during unbinding explicitly as devres
APIs are leveraged.  Otherwise kernel oops.

Signed-off-by: Weifeng Liu <weifeng.liu@intel.com>
---
 drivers/ivshm/ivshm_ivshmem.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/ivshm/ivshm_ivshmem.c b/drivers/ivshm/ivshm_ivshmem.c
index 69b2f1635aa9..44fda617f835 100644
--- a/drivers/ivshm/ivshm_ivshmem.c
+++ b/drivers/ivshm/ivshm_ivshmem.c
@@ -152,7 +152,6 @@ static void remove(struct pci_dev *pdev)
 	for (i = 0; i < idev->nvecs; i++)
 		free_irq(idev->msix_entries[i].vector, idev);
 
-	ivshm_unregister_device(&idev->info);
 	pci_release_regions(pdev);
 	pci_disable_device(pdev);
 }
-- 
2.17.1

